timemachine/cpp/src/context.cu

```diff
@@ -65,6 +65,16 @@ void Context::_verify_box(const double *box_buffer, cudaStream_t stream) {
             }
         }
     }
+
+    const double max_box_dim = *std::max_element(box_buffer, box_buffer + 3 * 3);
```

Comment:
Might be more precise to check for the diagonal elements only for now, since I don't think we use the off-diagonal box elements for anything.

Reply:
Only check the diagonal elements in https://github.com/proteneer/timemachine/pull/1401/commits/74585eccf9fb7955be8294ad049c367127b35180